server/src/main/java/org/elasticsearch/search/aggregations/metrics/CardinalityAggregator.java

```diff
@@ -61,6 +62,7 @@ public CardinalityAggregator(
         String name,
         ValuesSourceConfig valuesSourceConfig,
         int precision,
+        CardinalityAggregatorFactory.ExecutionMode executionMode,
```

Comment:
Should this be called `exeuctionHint`? Because `null` means "you pick?"

Reply:
Ah. I see you expect a refactor soon. Well, may as well wait on the rename until then anyway.

Reply:
I was considering an explicit "heuristic" mode instead of relying on the null.  And I'm not sure when I'm going to get to that refactor.  WDYT?

Reply:
I'd prefer a `HEURISTIC` mode over null, yeah.